llm/main.py

```diff
@@ -0,0 +1,51 @@
+from flask import Flask, jsonify, abort, request
+from llm_functions import get_summary_api_function, get_tags_api_function
+import json
+from firebase_admin import initialize_app
+from firebase_functions import https_fn
+
+initialize_app()
+app = Flask(__name__)
+
+
+def is_intersection(keys, required_keys):
+    return (keys & required_keys) == required_keys
+
+
+@app.route("/summary", methods=["POST"])
+def summary():
+    body = json.loads(request.data)
+    # We require bill_id, bill_title, bill_text to exist as keys in the POST
+    if not is_intersection(body.keys(), {"bill_id", "bill_title", "bill_text"}):
+        abort(404, description="requires bill_id, bill_title, and bill_text")
+
+    summary = get_summary_api_function(
+        body["bill_id"], body["bill_title"], body["bill_text"]
+    )
+
+    if summary["status"] in [-1, -2]:
+        abort(500, description="Unable to generate summary")
```

Comment:
The status field returns either -1 or -2 if it fails... This is probably somewhat unfortunate here because if this API ever changes I will not be able to know about it.

I'll add a note to `get_summary_api_function` but we don't really have types.

Reply:
Or, maybe I should just check that this value is negative? technically the return type is `int`